exercises/hamming/test/hamming_test.dart

```diff
@@ -1,84 +1,58 @@
-import 'package:test/test.dart';
 import 'package:hamming/hamming.dart';
+import 'package:test/test.dart';
+
+final hamming = Hamming();
 
 /// Test cases match one-to-one with those described
 /// in the problem specification found here:
 /// https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json
 void main() {
-  final Hamming hamming = new Hamming();
+  final mustEqual = predicate(
+      (ArgumentError e) => e is ArgumentError && e.message == 'left and right strands must be of equal length',
+      'an ArgumentError with the message "left and right strands must be of equal length"');
+  final noEmptyStrand = predicate((ArgumentError e) => e is ArgumentError && e.message == 'no strand must be empty',
```

Comment:
Same as `mustEqual`
```suggestion
  final noEmptyStrand = predicate((e) => e is ArgumentError && e.message == 'no strand must be empty',
```

Reply:
Because of how we have analysis_options setup (no implicit casting), I had to set the parameter to Error so it's at least a generic Error. Setting it to dynamic had the analyzer throw an error. The tests run well, but I don't want the analyzer confusing users.

Reply:
About your comment on analyzer confusing the user. Either of the following should be sufficient
```dart
predicate((Object e) => e is ArgumentError && e.message == '...', '...');
```
```dart
predicate<dynamic>((dynamic e) => e is ArgumentError && e.message == '...', '...');
```

Reply:
On the note of throwing, Do you think exercise tests should be responsible for differentiating between `Error` and `Exception`. Or should the mentor be doing that?

Reply:
`dynamic` doesn't work, but `Object` does. Would you prefer I use that? Since Hamming is a Core exercise, I would prefer a bit more hand-holding. So I would opt for leaving `Error` in. But I'm open to be persuaded otherwise.

> On the note of throwing, Do you think exercise tests should be responsible for differentiating between Error and Exception. Or should the mentor be doing that?
That's a good question, tracks will undergo an update to V3 within the year, so how concepts are introduced are important. If a test could reasonably incorporate both, I'd love for the Test Suite or README to include a blurb about it.

Otherwise, it's up to the student and mentor.

Reply:
Sure, as a core exercise, it makes sense to use a more specific type `Error`.